permissions/page.js

```diff
@@ -0,0 +1,41 @@
+function updatePermissions() {
+  browser.permissions.getAll()
+  .then((permissions) => {
+    document.getElementById('permissions').innerText = permissions.permissions.join(', ');
+  });
+}
+
+function processChange(event) {
+  let permission = event.target.dataset.permission;
+  let result = document.getElementById('result');
+  let permissionMethod = (
+    event.target.dataset.action === 'grant' ?
+    browser.permissions.request : browser.permissions.remove
+  );
+
+  try {
+    permissionMethod({permissions: [permission]})
```

Comment:
This is neat, but I wonder if it obscures the actual calls to the API, which are the main focus of this example. Just an opinion though :).